src/pal/src/misc/time.cpp

```diff
@@ -270,6 +275,29 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_CLOCK_MONOTONIC
+    {
+        kern_return_t machRet;
+        clock_serv_t clock;
+        mach_timespec_t mts;
+        host_t host = mach_host_self();
+
+        if((machRet = host_get_clock_service(host, SYSTEM_CLOCK, &clock)) != KERN_SUCCESS)
```

Comment:
>>> caching enabled it is around it is around 16.4x slower
It is better than before, but still ~100x slower compared to Windows...

I have looked around for what is available ... have you considered using mach_absolute_time()? It seems to be the fastest way to get what you want.

Also, I am wondering whether we need to make similar change in other places like https://github.com/dotnet/coreclr/blob/master/src/pal/src/synchmgr/synchmanager.cpp#L4881 to be robust around clock time adjustment (e.g. day light switch). This can be tracked as separate issue.